public/js/cin.js

```diff
@@ -0,0 +1,527 @@
+            <a
+              href="#liveStream"
+              class="ml-5 pl-5 fa fa-play-circle youtubePlay"
+              style="font-size:5em;color:white"
+            ></a>
+          </div>
+        </div>
+      </div>
+    </div>
+  `;
+};
+
+const PastScheduleCard = (props) => {
+  var playButton = "";
+  if (props.youtubeID != null) {
+    playButton = html`<a
+      href="javascript:void(0)"
+      onClick=${() => props.callback(props.youtubeID)}
+      class="ml-5 pl-5 fa fa-play-circle youtubePlay"
+      style="font-size:5em;color:white"
+    ></a>`;
+  }
+  return html`
+    <div
+      class="card bg-cin-card mx-auto m-2 mt-4 mb-4"
+      style="width: 35em;  max-width: 90%;"
+    >
+      <div class="card-body">
+        <div class="card-text"><h2>${props.position}</h2></div>
+        <div class="row">
+          <div class="col">
+            <div class="card-text"><h3>${props.candidate}</h3></div>
+            <div class="card-text">with <b>${props.interviewer}</b></div>
+            <div
+              class="card-text"
+              onClick=${() => props.callback(props.youtubeID)}
+            >
+              ${props.youtubeStatus}
+            </div>
+          </div>
+          <div class="col">${playButton}</div>
+        </div>
+      </div>
+    </div>
+  `;
+};
+
+function prettifyCandidates(candidates) {
+  var names = [];
+  candidates.forEach((candidate) => {
+    names.push(candidate.full_name);
+  });
+  return names.join(", ");
+}
+
+const getInterviewers = (event) => {
+  var names = [];
+  event.user_roles.forEach((user) => {
+    if (user.role.name == "Interviewer") {
+      names.push(user.user.name);
+    }
+  });
+  return names.join(", ");
+};
+
+const getLatestODVideo = () => {
+  // return "7lUz0xU5d9g";
+  for (let i = interviews.length - 1; i >= 0; i--) {
+    if (interviews[i].interview.youtube_id != null) {
+      return interviews[i].interview.youtube_id;
+    }
+  }
+  return defaultYouTube; // URY Ad (there's nothing else to show :()
+};
+
+function escapeRegExp(string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+const ScheduleArea = () => {
+  const [slots, setSlots] = useState([]);
+  const searchTerm = useRef("");
+  const [searched, setSearched] = useState(true);
+  const [youtubeVid, setYoutubeVid] = useState(getLatestODVideo());
+  const [youtubeTitle, setYoutubeTitle] = useState("You Just Missed");
+
+  const handleSearch = (event) => {
+    searchTerm.current = event.target.value;
+    setSearched(false);
+  };
+
+  const updateYoutube = (id) => {
+    setYoutubeVid(id);
+    setYoutubeTitle("");
+  };
+
+  const updateSchedule = (auto) => {
+    if (!auto || searchTerm.current == "") {
+      console.log("Update Schedule");
+      var tmp = [
+        html`<input
+          type="search"
+          id="search"
+          class="form-control mx-auto bg-cin"
+          placeholder="Search"
+          aria-label="Search"
+          onKeyUp=${handleSearch}
```

Comment:
Hkjdajskdaskjhkjdajskhkdjsadsa... just use a `const [searchTerm, setSearchTerm] = useState("")` and a `useEffect(() => {}, [searchTerm])`. Absolutely zero need for a ref here.

Reply:
tried that, didn't work